### PR TITLE
7094 carto api vis controller spec performance

### LINKED ISSUE #7094: Improve tests for Carto::Api::VisualizationsController
This test is too long, in fact it takes around ~15-20 minutes by itself. When run in the parallel environment, it dominates the execution time: all other tests are finished 8 minutes earlier than this one, even if this is the first test to start.

```
08:34:24 Finished: spec/requests/admin/organization_users_controller_spec.rb Port: 6003
08:34:34 Finished: spec/models/user_organization_spec.rb Port: 6002
08:34:45 Finished (FAILED): spec/requests/superadmin/users_spec.rb Port: 6005
08:35:00 Finished: spec/models/carto/user_spec.rb Port: 6015
08:43:54 Finished: spec/requests/carto/api/visualizations_controller_spec.rb Port: 6007
08:43:54 
08:43:54 real	19m25.708s
08:43:54 user	40m46.356s
08:43:54 sys	8m0.487s
```

Making this single test faster (probably by avoiding recreation of resources) will directly impact the execution time of tests.

Meta: https://github.com/CartoDB/cartodb/issues/6754

### PR DESCRIPTION
**Pull request by javitonino**
Closes #7094 

This removes a lot of user and tables generation from the Carto::Api::VisualizationsController tests, speeding up the tests. There is lots to do here, as this is a massive spec, but this addresses the low-hanging fruit.

After these changes, it still is the last test to finish by a minute, but it stills offers a noticeable improvement:
Sequential: 8m -> 3m
Parallel: 21m -> 15m (for whole suite), Total build ~55 -> ~45m.

### CODE REVIEW

**Comment by javitonino** [comment]
@gfiorav, can you CR?

File: spec/requests/carto/api/visualizations_controller_spec.rb [hunk 1]
```diff
@@ -79,82 +79,74 @@ def vizjson_vx_version
...
     include_context 'users helper'
 
-    before(:each) do
-      CartoDB::NamedMapsWrapper::NamedMaps.any_instance.stubs(:get => nil, :create => true, :update => true)
+    before(:all) do
+      CartoDB::NamedMapsWrapper::NamedMaps.any_instance.stubs(get: nil, create: true, update: true)
 
       @user_1 = FactoryGirl.create(:valid_user, private_tables_enabled: false)
-      @user_2 = FactoryGirl.create(:valid_user)
+      @table1 = create_random_table(@user_1)
```
Comment (gfiorav) [inline spec/requests/carto/api/visualizations_controller_spec.rb]: I generally prefer using `FactoryGirl` directly...

Any reason to use the method?

**Comment by gfiorav** [comment]
Made a minor comment, 📦 it after that

File: spec/requests/carto/api/visualizations_controller_spec.rb (hunk 1 above)
Comment (javitonino) [inline spec/requests/carto/api/visualizations_controller_spec.rb]: This is taken from the existing code, just moved it here so it is run only once. Is there a factory for tables? The one I could find doesn't work (throws an error on create) and does not do all the initialization needed by the test.

Comment (gfiorav) [inline spec/requests/carto/api/visualizations_controller_spec.rb]: If you lurked around 🌴 go for it!

**Pull request merged**